README.md

```diff
@@ -1 +1,199 @@
-# ocm-kubevirt-samples
\ No newline at end of file
+# ocm-kubevirt-samples
+
+This repo contains sample kubevirt resources for deployment in
+Kubernetes or OpenShift clusters.
+
+
+## Using in OpenShift
+
+Use ACM console to deploy the VM, enable DR or delete the
+VM.
+
+### Deploying the VM
+
+1. Create new application using:
+   - repo: this repo URL
+   - branch: main
+   - path: one of the `vm-*-odr-*` directories
+
+1. To enable DR assign DR Policy
+
+### Testing DR
+
+Use ACM console to perform `failover` or `relocate` actions.
+
+### Deleting the VM
+
+Use ACM console to delete the application.
+
+
+## Using in Kubernetes
+
+In Kubernetes you need to deploy the VM and DR resources using the
+command line tools or API.
+
+### Deploying the channel
+
+Deploy the channel to introduce this repo to *OCM*:
+
+```sh
+kubectl apply -k channel --context hub
+```
+
+### Deploying the VM subscription
+
+Create a secret with your public key to allow SSH access to the VM on
+both managed clusters:
+
+```sh
+for cluster in dr1 dr2; do
+    kubectl create ns kubevirt-sample --context $cluster
+    kubectl create secret generic my-public-key \
+        --from-file key1=$HOME/.ssh/id_rsa.pub \
+        --namespace kubevirt-sample \
+        --context $cluster
+done
+```
```

Comment:
Testing replacement by generating the secret in the kustomization:
https://github.com/nirs/ramen/commit/dd0e3ee605bec720670f72e6a198befc7bfa361e

With this change the only manual step is:

```
cp ~/.ssh/id_rsa.pub vm-standalone-pvc-k8s/
```

So this is trivial to customize for every user, and forgetting to copy the key will
fail early with a clear message. If we create some scripts to automated test this step can be automated.

I tried another approach - adding static key to the repo, but this causes false alarms
for the security team (got mail about possible leak of the private key) so it is not
the right direction.

Another approach is to automated creating new keypair for testing, but `virtctl
--identity-file` is broken (value is ignored when using the native ssh client) so
we need to expose the vm ssh port via a service (e.g. NodePort) and use the standard
ssh client. Seems too complicated so I'm starting with the simple approach.

@aglitke what do you think?

Reply:
Tested, it cannot work with OCM - the key must be in the remote git repo.
Posting a working solution soon.